### ISSUE TITLE
Cannot download differentially..: Error: Maximum allowed size is 5 MB

### ISSUE THREAD

**Issue by harryhahn** [electron-userland/electron-builder#3564]
* **Version**: 
  "electron-builder": "^20.38.4",
 "electron-updater": "^4.0.7",

* **Target**:  "target": "NSIS",
          "arch": [ "x64" ]

Old and new blockmaps are the same ??
Where can i define the Size which is mentioned in the following error "Cannot download differentially, fallback to full download: Error: Maximum allowed size is 5 MB**"

Checking for update
Found version 1.0.6 (url: dallasspur-setup-1.0.6.exe)
Downloading update from dallasspur-setup-1.0.6.exe
updater cache dir: C:\Users\harry\AppData\Local\dallasspur-updater
No cached update info available
**Download block maps (old: "https://api.github.com/repos/harryhahn/DallasTrack/releases/assets/10233924.blockmap", new: https://api.github.com/repos/harryhahn/DallasTrack/releases/assets/10233924.blockmap)
c
    at Object.responseHandler (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\src\httpExecutor.ts:207:26)
    at ClientRequest.request.createRequest.response (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\src\httpExecutor.ts:267:17)
    at ClientRequest.emit (events.js:182:13)
    at URLRequest.ClientRequest.urlRequest.on (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\electron.asar\browser\api\net.js:207:12)
    at URLRequest.emit (events.js:182:13)
From previous event:
    at ElectronHttpExecutor.downloadToBuffer (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\src\httpExecutor.ts:178:60)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:145:46
    at Generator.next (<anonymous>)
From previous event:
    at downloadBlockMap (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:144:29)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:166:85
    at Generator.next (<anonymous>)
From previous event:
    at NsisUpdater.differentialDownloadInstaller (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:134:173)
    at Object.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:34:42)
    at Generator.next (<anonymous>)
From previous event:
    at Object.task (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:31:11)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:609:25
From previous event:
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:544:68)
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\BaseUpdater.ts:27:18)
    at NsisUpdater.doDownloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:27:17)
    at NsisUpdater.downloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:419:19)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:382:49
    at Generator.next (<anonymous>)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
From previous event:
    at NsisUpdater.doCheckForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:359:34)
    at NsisUpdater.checkForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:220:35)
    at App.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\main.js:88:15)
    at App.emit (events.js:187:15)
Error: Error: Cannot download "https://api.github.com/repos/harryhahn/DallasTrack/releases/assets/10233924", status 404: Not Found
    at ClientRequest.request.createRequest.response (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\src\httpExecutor.ts:247:26)
    at ClientRequest.emit (events.js:182:13)
    at URLRequest.ClientRequest.urlRequest.on (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\electron.asar\browser\api\net.js:207:12)
    at URLRequest.emit (events.js:182:13)
From previous event:
    at ElectronHttpExecutor.download (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\electronHttpExecutor.ts:23:73)
    at Object.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:35:35)
From previous event:
    at Object.task (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:31:11)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:609:25
From previous event:
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:544:68)
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\BaseUpdater.ts:27:18)
    at NsisUpdater.doDownloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:27:17)
    at NsisUpdater.downloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:419:19)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:382:49
    at Generator.next (<anonymous>)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
From previous event:
    at NsisUpdater.doCheckForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:359:34)
    at NsisUpdater.checkForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:220:35)
    at App.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\main.js:88:15)
    at App.emit (events.js:187:15)
Unhandled rejection Error: Cannot download "https://api.github.com/repos/harryhahn/DallasTrack/releases/assets/10233924", status 404: Not Found
    at ClientRequest.request.createRequest.response (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\node_modules\builder-util-runtime\src\httpExecutor.ts:247:26)
    at ClientRequest.emit (events.js:182:13)
    at URLRequest.ClientRequest.urlRequest.on (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\electron.asar\browser\api\net.js:207:12)
    at URLRequest.emit (events.js:182:13)
From previous event:
    at ElectronHttpExecutor.download (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\electronHttpExecutor.ts:23:73)
    at Object.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:35:35)
From previous event:
    at Object.task (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:31:11)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:609:25
From previous event:
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:544:68)
    at NsisUpdater.executeDownload (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\BaseUpdater.ts:27:18)
    at NsisUpdater.doDownloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\NsisUpdater.ts:27:17)
    at NsisUpdater.downloadUpdate (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:419:19)
    at C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:382:49
    at Generator.next (<anonymous>)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
From previous event:
    at NsisUpdater.doCheckForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:359:34)
    at NsisUpdater.checkForUpdates (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\node_modules\electron-updater\src\AppUpdater.ts:220:35)
    at App.<anonymous> (C:\Users\harry\AppData\Local\Programs\dallasspur\resources\app.asar\main.js:88:15)
    at App.emit (events.js:187:15)

**Comment by grantdfoster** [comment]
Yes, I am having a similar issue with a 404 triggering on the fetch to the release asset

**Comment by rayonef** [comment]
Same issue, any luck ?

**Comment by orgertot** [comment]
I have the same issue... How to solve this?

**Comment by pmcarlos** [comment]
@orgertot what problem did you have? Error: Maximum allowed size is 5 MB or status 404: Not Found

**Comment by pmcarlos** [comment]
Logger keeps sending the error, but after making the repo public it updates...

**Comment by orgertot** [comment]
@pmcarlos Error: Maximum allowed size is 5 MB

**Comment by orgertot** [comment]
Pinging @develar 
This problem will effect more and more users because of githubs new free private repos..

**Comment by Littlebigdondon** [comment]
I, too, am hitting the 5 MB error on NSIS target updates from a private repo -- and making it public isn't an option. Anybody come across a workaround yet?

**Comment by PauliusKrutkis** [comment]
Same issue for me too

**Comment by lovely-man** [comment]
Same issue for me too ++

**Comment by melbin** [comment]
Exactly same issue here too.

**Comment by melbin** [comment]
I solved the problem downgrading the **electron-updater and builder** versions.

downgrade to the following versions:
electron: 4.0.6
electron-updater: 3.2.3
electron-builder: 20.34.0

The important thing of this, is that it works and your proyect on GitHub can continue being PRIVATE.

**Comment by lovely-man** [comment]
将electron-updater回退到3.2.3之后，提示需要升级到4.0才能使用 @melbin

**Comment by DoomyTheFroomy** [comment]
Unfortunately the downgrade to `electron-builder: 20.34.0` workaround is not a solution if you want to sign your target. Because you will run into another issue. Hope this will be solved soon.

**Comment by cuddeford** [comment]
We too are having this problem and downgrading the package versions had no effect.

**Comment by Littlebigdondon** [comment]
@melbin Downgrading electron-updater to 3.2.3 still fails to download differential updates, but does at least fall back to downloading the newest release _in full_. However, I don't get notifications when the updates are ready to install (only on Windows -- macOS still notifies just fine).

Note that downgrading electron-builder had no effect on this issue for me. 

@DoomyTheFroomy I was still able to sign all executables on a Mac (using [my solution](https://github.com/electron-userland/electron-builder/issues/1299#issuecomment-474613962) for code signing the Windows executable with an EV hardware token on macOS). What other issue did you run into?

**Comment by DoomyTheFroomy** [comment]
@Littlebigdondon yes exactly this error. I haven’t seen your workaround but I didn't wanted change here everything just for a workaround. Hopefully they are solving the issue pretty fast. 
**Offtopic** Perhaps someone can show an introduction which url's the generic endpoint have to support. I think the private github repository has too many issues.

**Comment by franckadil** [comment]
> I solved the problem downgrading the electron-updater and builder versions.
> downgrade to the following versions:
> electron: 4.0.6
> electron-updater: 3.2.3
> electron-builder: 20.34.0
> The important thing of this, is that it works and your proyect on GitHub can continue being PRIVATE.

I can confirm this works for my private repo.

**Comment by mr-islam** [comment]
My project repo is public, yet I still have this issue. Using:

```
"electron": "^4.2.8",
"electron-builder": "^21.1.5",
"electron-updater": "^4.1.2",
```

**Comment by vKarter** [comment]
I have still the issue 

> [2019-08-27 21:37:03.555] [error] Cannot download differentially, fallback to full download: Error: Maximum allowed size is 5 MB
```
"electron-builder": "^21.2.0",
"electron-updater": "^4.1.2",
"electron": "^6.0.2"
```

And I don't want to downgrade. Any solution?

**Comment by Dezainer** [comment]
Downgrading it solves the 5mb error, but results in another issue with publishing the bundle to Github(my repo is private as well). Any ideas ?

**Comment by zeejay09** [comment]
This happens to me also, I have a private repo.

**Comment by OldManMeta** [comment]
How is it that this error is not solved by now??

**Comment by NoahAndrews** [comment]
It should be solved, are you using the latest version of `electron-updater`?

**Comment by OldManMeta** [comment]
@NoahAndrews - thanks for your response. Yes I am using the latest version, but I think it may actually be due to my config. I noticed the other day someone had useDifferencital or something like that in their config - I've not had time to test it, and to be honest, the entire update solution had given me a migraine, so I'll return to it in the coming weeks when I'm calmer ;-)

I will try and dig out the config values that I saw, and test that first. If it's still happening, I'll open a new issue to clean up the thread history. Cheers
